### PR TITLE
GO-3853 Fix creationDate for get_started

### PR DESCRIPTION
**Pull request by KirillSto**
https://linear.app/anytype/issue/GO-3853/newly-created-accounts-still-have-incorrect-dates-in-their-objects

- We should remove lastModifiedDate from objects of get_started usecase, as this timestamp is set as creationDate of all objects on new account creation
- Introduce complex filter _'valueFromRelation'_ that allows to filter out objects by comparing values of two different relations

### CODE REVIEW

File: pkg/lib/database/filter.go [hunk 1]
```diff
@@ -845,3 +868,91 @@ func (i *FilterNestedNotIn) AnystoreFilter() query.Filter {
...
+type Anystore2ValuesComp struct {
+	Path1, Path2 []string
+	CompOp       query.CompOp
+	buf1, buf2   []byte
+}
+
+func (e *Anystore2ValuesComp) Ok(v *fastjson.Value) bool {
+	value1 := v.Get(e.Path1...)
+	value2 := v.Get(e.Path2...)
+	comp := bytes.Compare(encoding.AppendJSONValue(e.buf1[:0], value1), encoding.AppendJSONValue(e.buf2[:0], value2))
```
Comment (deff7) [inline pkg/lib/database/filter.go:916]: e.buf is always nil, you should set it to a new value

```
e.buf = encoding.AppendJSONValue(e.buf1[:0], value1)
```

File: util/pbtypes/pbtypes.go [hunk 2]
```diff
@@ -276,6 +276,16 @@ func GetStringListValue(v *types.Value) []string {
 	return nil
 }
 
+func GetStructValue(v *types.Value) *types.Struct {
```
Comment (deff7) [inline util/pbtypes/pbtypes.go:279]: `v.GetStructValue()`

File: pkg/lib/database/filter.go (hunk 1 above)
Comment (KirillSto) [reply pkg/lib/database/filter.go]: Thank you! Added init of each buffer before comparison

File: pkg/lib/database/filter_test.go [hunk 3]
```diff
@@ -26,6 +26,15 @@ func assertFilter(t *testing.T, f Filter, obj *types.Struct, expected bool) {
 	assert.Equal(t, expected, anystoreFilter.Ok(val))
 }
 
+func assertFilterNoParseValidation(t *testing.T, f Filter, obj *types.Struct, expected bool) {
```
Comment (deff7) [inline pkg/lib/database/filter_test.go:29]: change original function please, this parsing validation is not necessary

**Review by deff7 — APPROVED** [review]

File: pkg/lib/database/filter_test.go (hunk 3 above)
Comment (KirillSto) [reply pkg/lib/database/filter_test.go]: Yeah, removed parse validation from filter assertion

**Pull request merged**